browser/ui/views/frame/vertical_tab_strip_region_view.cc

```diff
@@ -315,6 +315,28 @@ void VerticalTabStripRegionView::SetState(State state) {
   PreferredSizeChanged();
 }
 
+VerticalTabStripRegionView::ScopedStateResetter
+VerticalTabStripRegionView::ExpandTabStripForDragging() {
+  if (state_ == State::kExpanded)
+    return {};
+
+  auto resetter = std::make_unique<base::ScopedClosureRunner>(
```

Comment:
Oh, this resetter looks cool!

Reply:
Yeah, it's perfect for this kind of mementor pattern!